com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -882,6 +891,57 @@ internal void InitializeSensors()
 #endif
         }
 
+        void InitializeActuators()
+        {
+            ActuatorComponent[] attachedActuators;
+            if (m_PolicyFactory.UseChildActuators)
+            {
+                attachedActuators = GetComponentsInChildren<ActuatorComponent>();
+            }
+            else
+            {
+                attachedActuators = GetComponents<ActuatorComponent>();
+            }
+
+            // Support legacy OnActionReceived
+            var param = m_PolicyFactory.BrainParameters;
+            switch (param.VectorActionSpaceType)
+            {
+                case SpaceType.Continuous:
+                    vectorActuator = new ContinuousVectorActuator(param.VectorActionSize);
+                    break;
+                case SpaceType.Discrete:
+                    vectorActuator = new DiscreteVectorActuator(param.VectorActionSize);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException($"Unknown SpaceType used:{param.VectorActionSpaceType}");
+            }
+
+            if (vectorActuator != null)
```

Comment:
We should allow vectorActuator to be null (if there are 0 actions in BrainParameters), in which case you'd want to make sure actuators is still created.